python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -430,9 +432,20 @@ def runDataRef(self, patchRefList, subdir="", cosmos=None):
             if self.config.doPlotSkyObjectsSky:
                 skyObjCatAll = unforced[unforced["merge_peak_sky"]].copy(deep=True)
             if self.config.doPlotSkyObjects:
-                goodSky = (unforced["merge_peak_sky"] & (unforced["base_InputCount_value"] > 0)
-                           & (unforced["deblend_nChild"] == 0) & ~unforced["base_PixelFlags_flag_edge"])
-                skyObjCat = unforced[goodSky].copy(deep=True)
+                baseGoodSky = (unforced["merge_peak_sky"] & (unforced["base_InputCount_value"] > 0)
+                               & ~unforced["base_PixelFlags_flag_edge"])
+                skyObjCat = unforced[baseGoodSky].copy(deep=True)
+                if "deblend_scarletFlux" in unforcedSchema:
+                    # Only include the non-model (i.e. not deblended) scarlet
+                    # sources.  Note that we include the "deblend_skipped" sky
+                    # sources since they are equivalent to the scarlet isolated
+                    # non-model (i.e. not deblended) sources.
+                    goodSky = ((skyObjCat["parent"] == 0) & (skyObjCat["deblend_nChild"] == 1))
+                    goodSky |= ((skyObjCat["parent"] == 0) & (skyObjCat["deblend_nChild"] == 0)
```

Comment:
This is dangerous, because if `(parent == 0) & (deblend_nChild == 0)` then this means that the deblender failed. While I think that it's unlikely that a true parent blend that failed would also have the `merge_peak_sky` bit set to true, I would double check the sources that made this cut just to make sure this isn't the source of some of the weird results that you're seeing in your analysis plots.

Reply:
I would also insert a TODO here to use the results of the `isDeblendedPrimary` flag, which will be consistent for both deblenders once DM-28542 is merged.

Reply:
See comments + plots on the ticket.